pkg/apis/numaflow/v1alpha1/pipeline_types.go

```diff
@@ -643,9 +647,9 @@ func (pls *PipelineStatus) MarkConfigured() {
 }
 
 // MarkNotConfigured the Pipeline has configuration.
-func (pls *PipelineStatus) MarkNotConfigured(reason, message string) {
+func (pls *PipelineStatus) MarkNotConfigured(reason, message string, generation int64) {
```

Comment:
Setting generation together with setting conditions looks pretty weird - during one reconciliation, this will lead to multiple generation settings. Shall we use a separated `setObservedGeneration` function, and do it when it's needed?